### PR TITLE
codec(ticdc): open protocol decoder support handle only message

### LINKED ISSUE #9443: open-protocol decoder support decode message only have handle key columns

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #9443 

### What is changed and how it works?

* open-protocol support decode message handle key only event

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: pkg/sink/codec/open/open_protocol_decoder.go [hunk 1]
```diff
@@ -15,126 +15,113 @@ package open
...
-	if err := b.decodeNextKey(); err != nil {
-		return 0, false, err
-	}
-	return b.nextKey.Type, true, nil
+	nextKey   *internal.MessageKey
+	nextEvent *model.RowChangedEvent
+
+	storage storage.ExternalStorage
+
+	upstreamTiDB *sql.DB
```
Comment (CharlesCheung96) [inline pkg/sink/codec/open/open_protocol_decoder.go:48]: Will ClaimCheck and OnlyHandleKey work together?

**Review by CharlesCheung96 — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**